agreement_dynamic/models/agreement_section.py

```diff
@@ -203,3 +196,18 @@ def _render_template(self, template_txt, model, res_ids, datas=False):
                 render_result = u""
             results[res_id] = render_result
         return results[res_ids[0]] or results
+
+    def write(self, values):
```

Comment:
the problem here is that this is called whenever we hit same on the agreement form view, and not on the agreement.section form view. Tom, do you have an idea about how to resolve this?

Reply:
*save

Reply:
Ah yes... that's probably because it's an inline form view and only actually writes it when the whole parent record is saved.

There are ways to make it save directly, but then you're be in trouble if you are creating an agreement and directly creating a section as well, because then the parent agreement does not yet exist when you want to save it.

I think for now let's work around this problem: can you make an extra action button called 'Validate' that when pressed, calls 'button_validate' which then tries to calculate dynamic_content?

Combined with an error on the full Preview that pinpoints the offending section when rendering, I think that should be enough for now.

Reply:
The QWeb exception we saw yesterday was not thrown into the _compute_dynamic_content method, but from the qweb template when calling t-raw=object.dynamic content.
So what I can do about this, is to redirect the preview action to first call the field, catch any errors when rendering via mako,  and show them as user error. Essentially the same thing we do now in write(), but with a button.

Reply:
Yea, although then you'd be rendering two times.

Similar effect would probably be to get `dynamic_content` as a string, and then pass it to the QWEB template as a string variable, and inserting that with `t-raw` instead of using `t-raw` on `object.dynamic_content` directly.

Reply:
Did you manage to resolve this issue? Still seeing `<t t-raw="s.dynamic_content" />`, but maybe you resolved it in another way.

Reply:
For this one I completely removed testing rendering in write(), and added button validate that should be pressed before preview. If nothing happens when pressing validate, then rendering is ok. Otherwise, a usererror will be thrown.